apps/etterlatte-brev-api/src/main/kotlin/no/nav/etterlatte/brev/model/bp/BarnepensjonBeregning.kt

```diff
@@ -45,10 +47,86 @@ internal fun barnepensjonBeregning(
         grunnbeloep = Kroner(grunnbeloep.grunnbeloep),
         beregningsperioder = beregningsperioder,
         sisteBeregningsperiode = beregningsperioder.maxBy { it.datoFOM },
-        trygdetid = trygdetid.map { it.fromDto(anvendtMetode, metodeFraGrunnlag, avdoede) },
+        trygdetid = mappedeTrygdetider,
         erForeldreloes = erForeldreloes,
         bruktTrygdetid =
-            trygdetid.find { it.ident == sisteBeregningsperiode.trygdetidForIdent }?.fromDto(anvendtMetode, metodeFraGrunnlag, avdoede)
+            mappedeTrygdetider.find { it.ident == sisteBeregningsperiode.trygdetidForIdent }
                 ?: throw ManglerAvdoedBruktTilTrygdetid(),
+        forskjelligTrygdetid = forskjelligTrygdetid,
+    )
+}
+
+data class IdentMedMetodeIGrunnlagOgAnvendtMetode(
+    val ident: String,
+    val beregningsMetodeFraGrunnlag: BeregningsMetode,
+    val beregningsMetodeAnvendt: BeregningsMetode,
+)
+
+fun mapRiktigMetodeForAnvendteTrygdetider(
+    trygdetid: List<TrygdetidDto>,
+    avdoede: List<Avdoed>,
+    beregningsperioder: List<Beregningsperiode>,
+): List<TrygdetidMedBeregningsmetode> {
+    val anvendteTrygdetiderIdenter =
+        beregningsperioder
+            .mapNotNull { periode ->
+                periode.trygdetidForIdent?.let {
+                    IdentMedMetodeIGrunnlagOgAnvendtMetode(
+                        ident = it,
+                        beregningsMetodeFraGrunnlag = periode.beregningsMetodeFraGrunnlag,
+                        beregningsMetodeAnvendt = periode.beregningsMetodeAnvendt,
+                    )
+                }
+            }.associateBy { it.ident }
+    if (anvendteTrygdetiderIdenter.isEmpty()) {
+        throw ManglerAvdoedBruktTilTrygdetid()
+    }
+    val fallbackMetode =
+        anvendteTrygdetiderIdenter[beregningsperioder.maxBy { it.datoFOM }.trygdetidForIdent]
+            ?: throw ManglerAvdoedBruktTilTrygdetid()
+
+    return trygdetid.map {
+        val releventMetode = anvendteTrygdetiderIdenter[it.ident] ?: fallbackMetode
```

Comment:
Ikke helt tydelig for meg hvordan fallback her blir korrekt. Kanskje legge ved en liten beskrivelse?

Reply:
Ja, det gir mening!